esque/controller/topic_controller.py

```diff
@@ -96,9 +96,23 @@ def _get_altered_config(self, topic: Topic) -> Dict[str, str]:
             altered_config[name] = value
         return altered_config
 
-    def delete_topic(self, topic: Topic):
-        future = self.cluster.confluent_client.delete_topics([topic.name])[topic.name]
-        ensure_kafka_future_done(future)
+    def delete_topic(self, topic: Topic) -> bool:
+        return self.delete_topics([topic])
+
+    def delete_topics(self, topics: List[Topic]) -> bool:
+        futures = self.cluster.confluent_client.delete_topics([topic.name for topic in topics])
```

Comment:
Looks like there is a way to tell Kafka only to return upon success or timeout. So this might be more efficient because we don't need to keep polling all topics to figure out if they got deleted.

```python
    def delete_topics(self, topics: List[Topic]) -> bool:
        futures = self.cluster.confluent_client.delete_topics([topic.name for topic in topics], operation_timeout=60)
        ensure_kafka_futures_done(list(futures.values()))
        return True
```

Maybe it's also better to iterate over the futures and check their results instead of using `ensure_kafka_futures_done`.
That way we could get more informative error handling. Not all topics might fail and the ones that do, might have some information as to why they failed.

Reply:
It does look somewhat faster, subjectively.